src/user_agent_parser.py

```diff
@@ -0,0 +1,24 @@
+"""
+This module parses all the input user-agent lines from the stdin.
+"""
+
+import sys
+from user_agents import parse
+
+
+def get_dict(user_agent_line):
+    """This function parses given user-agent line into a dict of Device, OS and Browser.
+
+    :param user_agent_line: user-agent line.
+    :type user_agent_line: str.
+    :return: dict.
+    """
+    parsed = parse(user_agent_line)
+    return {'Device': parsed.device.family, 'OS': parsed.os.family, 'Browser': parsed.browser.family}
+
+
+for line in sys.stdin:
```

Comment:
For tests simplification you better extract processing loop into separate function. Then you wont need that hell with source code reading and exec